### PR TITLE
Allow PATCH objects with nullable attribute (e.g. DateTime?)

### LINKED ISSUE #95: PATCH request with nullable attributes does not work
Hi, 
Thanks for this great library! 

When you are trying to PATCH a model with nullable attributes, it does not work. 

Model example: 

```csharp

public class TodoItem : Identifiable
{
    Attr("description")]
    public string Description { get; set; }

    [Attr("created-date")]
    public DateTime CreatedDate { get; set; }

    [Attr("achieved-date")]
    public DateTime? AchievedDate { get; set; }
}
```

1st PATCH Request example: Field is not sent 
PATCH http://localhost:5000/api/todoitem/1
```json
{
  "description": "New description"
}
```

2nd PATCH Request example: Field is sent 
PATCH http://localhost:5000/api/todoitem/1
```json
{
  "description": "New description",
  "achieved-date": "2017-04-25T21:41:29.033Z"
}
```

Actual behavior: Exception is returned from the API. 
Expected behavior:
- Field is ignored if not sent (1st example)
- Field is updated with new value if sent (2nd example)

_Gerfaut_

### PR DESCRIPTION
**Pull request by Gerfaut**
Closes #95.

Ready to merge, tests are running. 

Problem was located in the AttrAttribute class, while setting the new value on model attributes. It was crashing on nullable types (e.g DateTime?). 

Fixed as http://stackoverflow.com/questions/3531318/convert-changetype-fails-on-nullable-types. 

Thank you!

### CODE REVIEW

File: src/JsonApiDotNetCore/Models/AttrAttribute.cs [hunk 1]
```diff
@@ -26,10 +26,15 @@ public void SetValue(object entity, object newValue)
             var propertyInfo = entity
                 .GetType()
                 .GetProperty(InternalAttributeName);
-                
-            var convertedValue = Convert.ChangeType(newValue, propertyInfo.PropertyType);
```
Comment (jaredcnance) [inline src/JsonApiDotNetCore/Models/AttrAttribute.cs]: Thanks for the PR!

We do have a method called `TypeHelper.ConvertType` which is where type conversions should be processed, but looks like I missed this here. Can you try using that method instead and see if your tests still pass?

File: src/JsonApiDotNetCore/Models/AttrAttribute.cs [hunk 2]
```diff
@@ -26,10 +26,15 @@ public void SetValue(object entity, object newValue)
...
-            propertyInfo.SetValue(entity, convertedValue);
+
+			if (propertyInfo != null)
+			{
+				Type t = Nullable.GetUnderlyingType(propertyInfo.PropertyType) ?? propertyInfo.PropertyType;
+
+				var convertedValue = (newValue == null) ? null : Convert.ChangeType(newValue, t);
+
+				propertyInfo.SetValue(entity, convertedValue, null);
+			}
```
Comment (jaredcnance) [inline src/JsonApiDotNetCore/Models/AttrAttribute.cs]: some funkiness with your tab spacing throughout...not a big deal just a note

Comment (Gerfaut) [inline src/JsonApiDotNetCore/Models/AttrAttribute.cs]: Thanks for review! I fixed it 😄

File: src/JsonApiDotNetCore/Models/AttrAttribute.cs (hunk 1 above)
Comment (Gerfaut) [inline src/JsonApiDotNetCore/Models/AttrAttribute.cs]: Hi @jaredcnance, I tested with `TypeHelper.ConvertType` and all tests are green! 

While reading this method, I was wondering if it was not a good idea to replace 

```
if (type.GetTypeInfo().IsGenericType && type.GetGenericTypeDefinition().Equals(typeof(Nullable<>)))
                type = Nullable.GetUnderlyingType(type);
```

by 

`type = Nullable.GetUnderlyingType(type) ?? type;`

It changes nothing about the result (I think), but it is more easy to read (maybe subjective), WDYT? 

Thanks a lot for review!

Comment (jaredcnance) [inline src/JsonApiDotNetCore/Models/AttrAttribute.cs]: Yeah that looks fine to me and I agree the intent is clearer. Thanks again for the PR!

**Pull request merged**